src/python/pants/init/extension_loader.py

```diff
@@ -42,31 +43,92 @@ def load_backends_and_plugins(
     :param backends2: v2 backends to load.
     :param build_configuration: The BuildConfiguration (for adding aliases).
     """
+    used_deprecated_backends = set(backends1).intersection(
+        {
+            "pants.backend.codegen.antlr.java",
+            "pants.backend.codegen.antlr.python",
+            "pants.backend.codegen.jaxb",
+            "pants.backend.codegen.ragel.java",
+            "pants.backend.codegen.wire.java",
+        }
+    )
+    if used_deprecated_backends:
+        formatted_backends = "\n  * ".join(sorted(used_deprecated_backends))
+        toml_backends = ",\n  ".join(repr(backend) for backend in sorted(used_deprecated_backends))
+        toml_config = indent(f"[GLOBAL]\nbackend_packages.remove = [\n  {toml_backends},\n]", "  ",)
+        ini_backends = ",\n    ".join(repr(backend) for backend in sorted(used_deprecated_backends))
+        ini_config = indent(f"[GLOBAL]\nbackend_packages = -[\n    {ini_backends},\n  ]", "  ")
+        msg = (
+            "\nYou have activated the following backend packages, which are planned to be removed "
+            f"due to low usage:\n\n  * {formatted_backends}\n\nTo prepare for this "
+            f"change, add this to your `pants.toml`:\n\n{toml_config}\n\nOr add this to your "
+            f"`pants.ini`:\n\n{ini_config}\n\nIf you still depend on any of these backends, please "
+            "email us at pantsbuild@gmail.com or message us on Slack and we will keep the "
```

Comment:
Probably `pants-devel <pants-devel@googlegroups.com>` instead?

Reply:
I explained to Benjy that I went with this because some users might be hesitant to publically email the group that they're using a backend we're claiming is "unused" and now we're only going to keep it because of them. I'm sure not everyone thinks this way, but I suspect some will be hesitant/embarrassed due to basic human psychology.

Fine with me if you prefer we change to pants-devel, though.

Reply:
Mm, yea, I could see that. I don't think that it will be clear to the reader who this mail goes to though.

`pants-committers` might be a middle ground?

Reply:
Benjy and I went with `pants-devel` under the argument that communication should be open, e.g. for transparency that we are actually listening to the user and not sweeping things under the rug.